File: addons/guide-cpp/src/triggers/guide_trigger_released.cpp
#include "guide_trigger_released.h"

using namespace godot;

void GUIDETriggerReleased::_bind_methods() {
}

GUIDETriggerReleased::GUIDETriggerReleased() {
}

GUIDETriggerReleased::~GUIDETriggerReleased() {
}

bool GUIDETriggerReleased::is_same_as(const Ref<GUIDETrigger> &other) const {
    Ref<GUIDETriggerReleased> o = other;
    if (o.is_null()) return false;
    return true; 
}

GUIDETrigger::GUIDETriggerState GUIDETriggerReleased::_update_state(Vector3 input, double delta, GUIDEAction::GUIDEActionValueType value_type) {
    if (!_is_actuated(input, value_type)) {
        if (_is_actuated(get_last_value(), value_type)) {
            return TRIGGERED;
        }
    }
    return NONE;
}

String GUIDETriggerReleased::_editor_name() const {
    return "Released";
}

String GUIDETriggerReleased::_editor_description() const {
    return "Fires once, when the input goes from actuated to not actuated. The opposite of the Pressed trigger.";
}

File: addons/GuideCS/Triggers/GUIDETriggerReleased.cs
using Godot;

namespace Guide;

[Tool]
[GlobalClass]
public partial class GUIDETriggerReleased : GUIDETrigger {
	public override bool IsSameAs(GUIDETrigger other) => other is GUIDETriggerReleased;

	public override GUIDETriggerState _UpdateState(Vector3 input, float delta, GUIDEAction.GUIDEActionValueType valueType) {
		if (!_IsActuated(input, valueType)) {
			if (_IsActuated(_LastValue, valueType)) return GUIDETriggerState.Triggered;
		}
		return GUIDETriggerState.None;
	}

	public override string _EditorName() => "Released";

	public override string _EditorDescription() => "Fires once, when the input goes from actuated to not actuated. The opposite of the Pressed trigger.";
}
